### PR TITLE
Convert generic easyblocks to `run_shell_cmd`

### PR DESCRIPTION
**Pull request by bartoldeman**
For ConfigureMake the default for the `verbose=` parameter for `build_step` is reversed in line with the default for `run_shell_cmd`.

### CODE REVIEW

File: easybuild/easyblocks/generic/configuremake.py [hunk 1]
```diff
@@ -323,11 +323,11 @@ def configure_step(self, cmd_prefix=''):
...
         )
 
-        (out, _) = run_cmd(cmd, log_all=True, simple=False)
+        res = run_shell_cmd(cmd)
 
-        return out
+        return res.output
 
-    def build_step(self, verbose=False, path=None):
+    def build_step(self, verbose=True, path=None):
```
Comment (boegel) [inline easybuild/easyblocks/generic/configuremake.py:330]: We should keep the `verbose=False` here, and use `not verbose` below?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/generic/configuremake.py [hunk 2]
```diff
@@ -351,7 +351,8 @@ def build_step(self, verbose=False, path=None):
             ])
             self.log.info("Building target '%s'", target)
 
-            (out, _) = run_cmd(cmd, path=path, log_all=True, simple=False, log_output=verbose)
+            res = run_shell_cmd(cmd, work_dir=path, output_file=verbose)
```
Comment (boegel) [inline easybuild/easyblocks/generic/configuremake.py:354]: `run_shell_cmd` has a different default to always log output in temporary files, and we want to keep doing that also for `ConfigureMake.build_step`

So we should:

* deprecate the `verbose` option for `ConfigureMake.build_step`
* not pass down `verbose` to `run_shell_cmd` at all

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/generic/configuremake.py (hunk 2 above)
Comment (bartoldeman) [reply easybuild/easyblocks/generic/configuremake.py]: Done!

**Comment by bartoldeman** [comment]
All generic easyblocks have been converted. Time for testing...

**Comment by boegel** [comment]
@SebastianAchilles Can you look into testing this a bit, as soon as https://github.com/easybuilders/easybuild-framework/pull/4423 is merged?

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _GCC-system.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
jscclxl1.int.jsc-clx.fz-juelich.de - Linux Rocky Linux 9.3, x86_64, Intel Xeon Processor (Cascadelake) (cascadelake), Python 3.9.18
See https://gist.github.com/SebastianAchilles/4d8f88f630344f13ce8be3286e6a10fe for a full test report.

**Comment by SebastianAchilles** [comment]
I think it makes sense to test, if possible, one EasyConfig per modified EasyBlock. To find recent EasyConfigs for an EasyBlock `<EasyBlock>` I used:
```
find . -type f | xargs grep -l "easyblock = '<EasyBlock>'" | xargs ls -rt | tail -20
```

I intend to test the following EasyConfigs:

| EasyBlock | Corresponding EasyConfig to test |
| --- | --- |
| `Binary` | [`Nextflow-23.10.0.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/n/Nextflow/Nextflow-23.10.0.eb) |
| `Cargo` | [`cryptography-41.0.1-GCCcore-12.3.0.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/c/cryptography/cryptography-41.0.1-GCCcore-12.3.0.eb) |
| `Cargo` | [`Evcxr-REPL-0.14.2-GCCcore-12.2.0-Rust-1.65.0.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/e/Evcxr-REPL/Evcxr-REPL-0.14.2-GCCcore-12.2.0-Rust-1.65.0.eb) |
| `CMakeMake` | [`RapidJSON-1.1.0-20230928-GCCcore-12.3.0.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/r/RapidJSON/RapidJSON-1.1.0-20230928-GCCcore-12.3.0.eb) |
| `CmdCp` | [`Ninja-1.11.1-GCCcore-13.2.0.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/n/Ninja/Ninja-1.11.1-GCCcore-13.2.0.eb) |
| `Conda` | [`TRUST4-1.0.5.1.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/t/TRUST4/TRUST4-1.0.5.1.eb) |
| `ConfigureMake` | [`strace-6.6-GCCcore-12.2.0.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/s/strace/strace-6.6-GCCcore-12.2.0.eb) | 
| `ConfigureMakePythonPackage` | [`MUSCLE3-0.7.0-foss-2022b.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/m/MUSCLE3/MUSCLE3-0.7.0-foss-2022b.eb) |
| `FortranPythonPackage` | _using derived EasyBlock numpy_ [`SciPy-bundle-2023.11-gfbf-2023b.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/s/SciPy-bundle/SciPy-bundle-2023.11-gfbf-2023b.eb) |
| `GoPackage` | [`SeqKit-2.3.1.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/s/SeqKit/SeqKit-2.3.1.eb) |
| `IntelBase` | _using derived EasyBlock Advisor_ [`Advisor-2023.2.0.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/a/Advisor/Advisor-2023.2.0.eb) |
| `JuliaPackage` | [`ResistanceGA-4.2-5-foss-2022a-R-4.2.1-Julia-1.9.2.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/r/ResistanceGA/ResistanceGA-4.2-5-foss-2022a-R-4.2.1-Julia-1.9.2.eb) |
| `MesonNinja` | [`libGLU-9.0.3-GCCcore-13.2.0.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/l/libGLU/libGLU-9.0.3-GCCcore-13.2.0.eb) |
| `OCamlPackage` | no easyconfig found |
| `OctavePackage` | [`Octave-7.1.0-foss-2021b.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/o/Octave/Octave-7.1.0-foss-2021b.eb) |
| `PerlModule` | [`Bio-SearchIO-hmmer-1.7.3-GCC-11.2.0.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/b/Bio-SearchIO-hmmer/Bio-SearchIO-hmmer-1.7.3-GCC-11.2.0.eb) |
| `PythonPackage` | [`Pillow-10.0.0-GCCcore-12.3.0.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/p/Pillow/Pillow-10.0.0-GCCcore-12.3.0.eb) |
| `RPackage` | [`castor-1.7.11-foss-2022a.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/c/castor/castor-1.7.11-foss-2022a.eb) |
| `RPM` | no easyconfig found |
| `RubyGem` | [`Ruby-3.2.2-GCCcore-12.2.0.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/r/Ruby/Ruby-3.2.2-GCCcore-12.2.0.eb) |
| `SCons` | [`Cantera-3.0.0-foss-2023a.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/c/Cantera/Cantera-3.0.0-foss-2023a.eb) |
| `SystemCompiler` | [`GCC-system.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/g/GCC/GCC-system.eb) |
| `SystemMPI` | dependencies not installed |
| `Tarball` | [`VSCode-1.85.0.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/v/VSCode/VSCode-1.85.0.eb) |
| `VersionIndependentPythonPackage` | [`JUBE-2.4.1.eb`](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/j/JUBE/JUBE-2.4.1.eb) |
| `Waf` | no easyconfig found |

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Nextflow-23.10.0.eb_ 
 * **SUCCESS** _cryptography-41.0.1-GCCcore-12.3.0.eb_ 
 * **SUCCESS** _Evcxr-REPL-0.14.2-GCCcore-12.2.0-Rust-1.65.0.eb_ 
 * **SUCCESS** _RapidJSON-1.1.0-20230928-GCCcore-12.3.0.eb_ 
 * **SUCCESS** _Ninja-1.11.1-GCCcore-13.2.0.eb_ 
 * **SUCCESS** _TRUST4-1.0.5.1.eb_ 
 * **SUCCESS** _strace-6.6-GCCcore-12.2.0.eb_ 
 * **SUCCESS** _MUSCLE3-0.7.0-foss-2022b.eb_ 
 * **SUCCESS** _SciPy-bundle-2023.11-gfbf-2023b.eb_ 
 * **SUCCESS** _SeqKit-2.3.1.eb_ 
 * **SUCCESS** _Advisor-2023.2.0.eb_ 
 * **SUCCESS** _ResistanceGA-4.2-5-foss-2022a-R-4.2.1-Julia-1.9.2.eb_ 
 * **SUCCESS** _libGLU-9.0.3-GCCcore-13.2.0.eb_ 
 * **SUCCESS** _Bio-SearchIO-hmmer-1.7.3-GCC-11.2.0.eb_ 
 * **SUCCESS** _Pillow-10.0.0-GCCcore-12.3.0.eb_ 
 * **SUCCESS** _castor-1.7.11-foss-2022a.eb_ 
 * **SUCCESS** _Ruby-3.2.2-GCCcore-12.2.0.eb_ 
 * **SUCCESS** _Cantera-3.0.0-foss-2023a.eb_ 
 * **SUCCESS** _GCC-system.eb_ 
 * **SUCCESS** _VSCode-1.85.0.eb_ 
 * **SUCCESS** _JUBE-2.4.1.eb_ 

Build succeeded for 21 out of 21 (21 easyconfigs in total)
jscclxc1.int.jsc-clx.fz-juelich.de - Linux Rocky Linux 9.3, x86_64, Intel Xeon Processor (Cascadelake) (cascadelake), Python 3.9.18
See https://gist.github.com/SebastianAchilles/d676bcc8bde9132013cb2c0d11476d3e for a full test report.

**Review by SebastianAchilles — APPROVED** [review]
lgtm

**Comment by SebastianAchilles** [comment]
Going in, thanks @bartoldeman!

File: easybuild/easyblocks/generic/pythonpackage.py [hunk 3]
```diff
@@ -811,13 +812,13 @@ def test_step(self, return_output_ec=False):
...
                 # print Python search path (just debugging purposes)
-                run_cmd("%s -c 'import sys; print(sys.path)'" % self.python_cmd, verbose=False, trace=False)
+                run_shell_cmd("%s -c 'import sys; print(sys.path)'" % self.python_cmd, hidden=True)
 
                 abs_pylibdirs = [os.path.join(actual_installdir, pylibdir) for pylibdir in self.all_pylibdirs]
                 extrapath = "export PYTHONPATH=%s &&" % os.pathsep.join(abs_pylibdirs + ['$PYTHONPATH'])
 
                 cmd = self.compose_install_command(test_installdir, extrapath=extrapath)
-                run_cmd(cmd, log_all=True, simple=True, verbose=False)
+                run_shell_cmd(cmd, hidden=True)
```
Comment (boegel) [inline easybuild/easyblocks/generic/pythonpackage.py:821]: @bartoldeman Did you intentionally hide the installation command before running the tests in `PythonPackage.test_step`?

Just wondering, not opposing it, sort of makes sense, but it is different from what we had before (`verbose=False` in `run_cmd` is not equivalent with `hidden=True` in `run_shell_cmd`)

**Pull request merged**